core/arch/arm/kernel/kern.ld.S

```diff
@@ -353,3 +381,48 @@ SECTIONS
 	}
 
 }
+
+/*
+ * This builds __flatmap_xxx_rx_start/_size to setup the TEE_RAM mapping.
+ * It exports the layout through __vcore_XXX_rX_start/_size.
+ *
+ *               /rx           Expected     __flatmap_unpg_rx_start / +size
+ * __vcore_/unpg_|ro_/start    ordering: |  __flatmap_unpg_ro_start / +size
+ *         \init \rw \size               |  __flatmap_unpg_rw_start / +size
+ *                                       v  __flatmap_init_rx_start / +size
+ *                                          __flatmap_init_ro_start / +size
```

Comment:
It took me a few seconds to parse that ;) in fact I don't think this comment has much value, it just repeats what can be seen in the file. Same for sunxi.